Classes/Service/LanguageFileService.php

```diff
@@ -379,7 +384,9 @@ protected function loadLanguageRecordsFromDatabase() {
 		$select = 'flag';
 		$from = 'sys_language';
 		$where = '1=1' . $cObj->enableFields('sys_language');
-		$sysLanguages = $GLOBALS['TYPO3_DB']->exec_SELECTgetRows($select, $from, $where);
+		/** @var \TYPO3\CMS\Core\Database\DatabaseConnection $db */
+		$db = &$GLOBALS['TYPO3_DB'];
```

Comment:
Please move this to a `protected function getDatabaseConnection()` so that we can mock this easily in unit tests. Also, no need to use reference - the `TYPO3_DB` global variable is an object and will be a reference by default.